File: Week_04/515/find_largest_value_in_each_tree_row.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func largestValues(root *TreeNode) []int {
	res := make([]int, 0)
	if root == nil {
		return res
	}
	queue := []*TreeNode{root}

	for len(queue) > 0 {
		n := len(queue)
		val := queue[0].Val
		for i := 0; i < n; i++ {
			if queue[i].Val > val {
				val = queue[i].Val
			}
			if queue[i].Left != nil {
				queue = append(queue, queue[i].Left)
			}
			if queue[i].Right != nil {
				queue = append(queue, queue[i].Right)
			}
		}
		queue = queue[n:]
		res = append(res, val)
	}
	return res
}

File: Week_04/515/find_largest_value_in_each_tree_row.py
from typing import List


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def largestValues(self, root: TreeNode) -> List[int]:
        res = []
        if root == None:
            return res

        queue = [root]
        while queue:
            n = len(queue)
            val = queue[0].val
            for i in range(n):
                if queue[i].val > val:
                    val = queue[i].val
                if queue[i].left is not None:
                    queue.append(queue[i].left)
                if queue[i].right is not None:
                    queue.append(queue[i].right)
            res.append(val)
            queue = queue[n:]
        return res
